docs/usage/types/bytesize.md

```diff
@@ -1,32 +1,49 @@
 # Bytes and ByteSize
 
 `bytes`
-: `bytes` are accepted as-is, `bytearray` is converted using `bytes(v)`, `str` are converted using `v.encode()`,
-  and `int`, `float`, and `Decimal` are coerced using `str(v).encode()`
+: `bytes` are accepted as-is, `bytearray` is converted using `bytes(v)`, and `str` is converted using `v.encode()`.
 
 `SecretBytes`
-: bytes where the value is kept partially secret; see [Secrets](secrets.md)
-
-`conbytes`
-: type method for constraining bytes
+: Like `bytes`, but where the value is kept partially secret; see [Secrets](secrets.md).
 
 `ByteSize`
-: converts a bytes string with units to bytes
+: Converts a string representing a number of bytes with units (such as `'1KB'` or `'11.5MiB'`) into an integer.
+
+`conbytes`
+: A function which returns a type that can be used to add constraints to `bytes`.
 
 ### Arguments to `conbytes`
 The following arguments are available when using the `conbytes` type function
 
-- `strip_whitespace: bool = False`: removes leading and trailing whitespace
-- `to_upper: bool = False`: turns all characters to uppercase
-- `to_lower: bool = False`: turns all characters to lowercase
 - `min_length: int = None`: minimum length of the byte string
 - `max_length: int = None`: maximum length of the byte string
 - `strict: bool = False`: controls type coercion
 
+However, for the sake of improved integration with type checkers, we now discourage the use of `conbytes` and other
+function calls used to return constrained types. Instead, you can use `pydantic.types.Strict` and `annotated_types.Len`
+as annotations to achieve these constraints:
+
+```py
+import annotated_types
+from typing_extensions import Annotated
+
+from pydantic import BaseModel, Strict
+
+
+class MyModel(BaseModel):
+    # Instead of `my_bytes: conbytes(strict=True, min_length=10, max_length=20)`, use:
+    my_bytes: Annotated[bytes, Strict(), annotated_types.Len(10, 20)]
+```
+
+
 ## Using ByteSize
 
-You can use the `ByteSize` data type to convert byte string representation to
-raw bytes and print out human readable versions of the bytes as well.
+You can use the `ByteSize` data type to (case-insensitively) convert a string representation of a number of bytes into
+an integer, and also to print out human-readable strings representing a number of bytes.
+
+We interpret `'1KB'` to mean 1000 bytes, and `'1KiB'` to mean 1024 bytes. In general, including a middle `'i'`
```

Comment:
> including a middle `'i'`

Well, yes but it sounds hacky to me as I know that those are just from an ISO standard and I would like to see a guarantee that this follows the `IEC 80000-13` standard.

Reply:
@lig do you have a link we can use for reference about this? I couldn't find/come up with a good way to describe it, would much rather defer to someone else's description. Open to suggestions here if you want to take a stab.

Reply:
I think the closest link that doesn't point to a paywall directly is to the Wikipedia page https://en.wikipedia.org/wiki/ISO/IEC_80000

Reply:
```suggestion
In conformance with [IEC 80000-13 Standard](https://en.wikipedia.org/wiki/ISO/IEC_80000) we interpret `'1KB'` to mean 1000 bytes, and `'1KiB'` to mean 1024 bytes. In general, including a middle `'i'`
```